### DISCUSSION TITLE
How to do Route Guard with Nuxt 3?

### DISCUSSION THREAD

**Discussion by remiconnesson** [nuxt/nuxt#18188] [General]
Hey, newbie to nuxt and vue here, starting directly with vue3 and nuxt3.

I'm experimenting with nuxt3 by porting one of my vue3 project, and I'havent been able to find an answer to the following question in the documentation:

How can I include a beforeEnter intercept with nuxt3 as there is no router file?

I have found this answer for nuxt2 https://nuxtjs.org/docs/components-glossary/middleware/ but didn't find any mention of front end "middleware" in the documentation.

I know I must be missing something but I don't know what, can you help me in the right direction please?

**Comment by lguegu50** [comment]
Hello remi,

Same question here.
I've worked on a work around with using the layout (putting the middleware behaviour in the setup of the layout) but it doesn't working as expected because the page is displayed a short moment before the behaviour (a redirect on my side) is applied.

Has anyone a better solution ?

Thanks.

**Comment by eMeRiKa13** [comment]
There is another post with a solution for this problem: https://github.com/nuxt/framework/discussions/1722

**Reply by jmarshall9120** [reply]
Link is dead

**Comment by jjmoorhouse** [comment]
Hello, did you find a solution to this? We're currently suffering from this issue and the other post with the solution has been removed

**Reply by remiconnesson** [reply]
I think client side route guard has been implemented in the framework, so you should be able to implement those by looking in the doc. I don't recall top of mind what the feature is

**Reply by undermoonn** [reply]
https://nuxt.com/docs/guide/directory-structure/middleware#usage

**Reply by remiconnesson** [reply]
Yup that's the one thank you @undermoonn

**Comment by gmbeke** [comment]
Try some tricks written in this link will help you to have a clear idea on how to implement the guard from the level of page, and global guards

https://masteringnuxt.com/blog/how-to-create-navigation-guards-in-nuxt-with-definenuxtroutemiddleware
